src/client/config.rs

```diff
@@ -0,0 +1,51 @@
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use std::fs::{create_dir_all, File};
+use std::io::{BufReader, BufWriter, Write};
+use std::path::PathBuf;
+
+#[derive(Serialize, Deserialize)]
+pub struct ClientConfig {
+    version: u64,
+    pub authentication: HashMap<String, String>,
+    pub default_url: Option<String>,
+}
+
+impl ClientConfig {
+    pub fn new() -> ClientConfig {
```

Comment:
Should `new` receive the version as a param or should we leave it as is and change it in the future if needed?

Reply:
Good question! My intuition is no, since the version is a function of the schema/format of the encoded JSON, and should stay close to it. I'm adding it here now in case we ever need to change the schema/format, so that we can detect an older version of the configuration file and migrate it. The caller shouldn't have any knowledge of this internal detail of the configuration file. For example, you wouldn't want a caller to say that the configuration format is a v2 format when it is in fact v1. Let me know if that makes sense.

Reply:
I think you're right, I'll mark this as resolved :)